src/Nethermind/Nethermind.Network/SessionMonitor.cs

```diff
@@ -150,14 +138,18 @@ private void StartPingTimer()
                 try
                 {
                     _pingTimer.Enabled = false;
-                    SendPingMessages();
+                    SendPingMessagesAsync().ContinueWith(x =>
```

Comment:
Recursion safe here?

Reply:
Timer stopped, and not re-enabled to completion, shouldn't be recursion?